include/alp/omp/io.hpp

```diff
@@ -0,0 +1,138 @@
+		"* Possible fix 1 | Remove no_casting from the template parameters "   \
+		"in this call to " y ".\n"                                             \
+		"* Possible fix 2 | Provide a value that matches the expected type.\n" \
+		"********************************************************************" \
+		"********************************************************************" \
+		"******************************\n" );
+
+namespace alp {
+
+	/**
+	 * Sets all elements of the given matrix to the value of the given scalar.
+	 * C = val
+	 *
+	 * @tparam descr
+	 * @tparam OutputType      Data type of the output matrix C
+	 * @tparam OutputStructure Structure of the matrix C
+	 * @tparam OutputView      View type applied to the matrix C
+	 * @tparam InputType       Data type of the scalar a
+	 *
+	 * @param C    Matrix whose values are to be set
+	 * @param val  The value to set the elements of the matrix C
+	 *
+	 * @return RC  SUCCESS on the successful execution of the set
+	 */
+	template< Descriptor descr = descriptors::no_operation,
+		typename OutputType, typename OutputStructure, typename OutputView, typename OutputImfR, typename OutputImfC,
+		typename InputType, typename InputStructure
+	>
+	RC set(
+		Matrix< OutputType, OutputStructure, Density::Dense, OutputView, OutputImfR, OutputImfC, omp > &C,
+		const Scalar< InputType, InputStructure, omp > &val
+	) noexcept {
+
+		static_assert(
+			!std::is_same< OutputType, void >::value,
+			"alp::set (set to matrix): cannot have a pattern matrix as output"
+		);
+#ifdef _DEBUG
+		std::cout << "Called alp::set (matrix-to-value, omp)" << std::endl;
+#endif
+		// static checks
+		NO_CAST_ASSERT(
+			( !( descr & descriptors::no_casting ) || std::is_same< InputType, OutputType >::value ),
+			"alp::set", "called with non-matching value types"
+		);
+
+		static_assert(
+			!internal::is_functor_based<
+				Matrix< OutputType, OutputStructure, Density::Dense, OutputView, OutputImfR, OutputImfC, omp >
+			>::value,
+			"alp::set cannot be called with a functor-based matrix as a destination."
+		);
+
+		if( !internal::getInitialized( val ) ) {
+			internal::setInitialized( C, false );
+			return SUCCESS;
+		}
+
+		const Distribution &d = internal::getAmf( C ).getDistribution();
+
+		RC rc = SUCCESS;
+
+		#pragma omp parallel for
+		for( size_t thread = 0; thread < config::OMP::current_threads(); ++thread ) {
+			const size_t tr = d.getThreadCoords( thread ).first;
+			const size_t tc = d.getThreadCoords( thread ).second;
+			const auto block_grid_dims = d.getLocalBlockGridDims( tr, tc );
+
+			RC local_rc = SUCCESS;
+
+			for( size_t br = 0; br < block_grid_dims.first; ++br ) {
+				for( size_t bc = 0; bc < block_grid_dims.second; ++bc ) {
+
+					// Get a reference matrix view over the block
+					auto refC = internal::get_view( C, tr, tc, 1 /* rt */, br, bc );
+
+					// Construct a reference Scalar container from the input Scalar
+					Scalar< InputType, InputStructure, reference > ref_val( *val );
```

Comment:
Perhaps we should have have a way to construct from alp::Scalar with different backend to avoid going through access operator?